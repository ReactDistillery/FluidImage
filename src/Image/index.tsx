import * as React from "react";

export interface ImageProps {
  src: string;
  alt?: string;
  title?: string;
  hasImageLoaded?: boolean;
  onLoad: Function;
}

const styles: React.CSSProperties = {
  image: {
    maxWidth: "100%",
    maxHeight: "100%",
    position: "relative",
    zIndex: 1,
    transition: "opacity 0.6s"
  }
};

const Image = ({
  onLoad,
  hasImageLoaded = false,
  title,
  alt,
  src
}: ImageProps) => (
  <img
    onLoad={event => onLoad(event)}
    src={src}
    title={title}
    style={{
      ...styles.image,
      opacity: Number(hasImageLoaded)
    }}
    alt={alt}
    draggable={false}
  />
);

export default Image;
