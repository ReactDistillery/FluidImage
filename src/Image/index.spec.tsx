import React from "react";
import { create } from "react-test-renderer";
import Image from "./";

const minimalProps = {
  src: "testImage.jpeg",
  onLoad: () => {}
};

const fullProps = {
  src: "testImage.jpeg",
  alt: "some description",
  title: "some title",
  hasImageLoaded: true,
  onLoad: () => {}
};

describe("FluidImage - Image", () => {
  test("Renders an img with minimal props", () => {
    const image = create(<Image {...minimalProps} />).toJSON();
    expect(image).toMatchSnapshot();
  });
  test("Renders an img with full props", () => {
    const image = create(<Image {...fullProps} />).toJSON();
    expect(image).toMatchSnapshot();
  });
  test("it calls onLoad when image is loaded", () => {
    const onLoad = jest.fn();
    const image = create(<Image onLoad={onLoad} src="" />);
    image.root.findByType("img").props.onLoad();
    expect(onLoad.mock.calls.length).toBe(1);
  });
});
