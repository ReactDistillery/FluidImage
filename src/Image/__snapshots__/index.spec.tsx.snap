// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FluidImage - Image Renders an img with full props 1`] = `
<img
  alt="some description"
  draggable={false}
  onError={[Function]}
  onLoad={[Function]}
  src="testImage.jpeg"
  style={
    Object {
      "maxHeight": "100%",
      "maxWidth": "100%",
      "opacity": 1,
      "position": "relative",
      "transition": "opacity 0.6s",
      "zIndex": 1,
    }
  }
  title="some title"
/>
`;

exports[`FluidImage - Image Renders an img with minimal props 1`] = `
<img
  alt={undefined}
  draggable={false}
  onError={[Function]}
  onLoad={[Function]}
  src="testImage.jpeg"
  style={
    Object {
      "maxHeight": "100%",
      "maxWidth": "100%",
      "opacity": 0,
      "position": "relative",
      "transition": "opacity 0.6s",
      "zIndex": 1,
    }
  }
  title={undefined}
/>
`;
